test/Conversion/VectorToXsmm/vector-to-transpose.mlir

```diff
@@ -0,0 +1,129 @@
+    vector.transfer_write %1, %arg1[%c0, %c0, %c0] {in_bounds = [true, true, true]} : vector<5x5x3xf32>, memref<5x5x3xf32>
+    return
+}
+// CHECK-LABEL: func.func @transpose_op_1(
+// CHECK: %[[arg0:.*]]: memref<5x3x5xf32>, %[[arg1:.*]]: memref<5x5x3xf32>) {
+// CHECK-NOT: call @xsmm_unary_dispatch
+// CHECK-NOT: call @xsmm_unary_invoke
+
+// -----
+func.func @vnni_packing_0(%arg0: memref<32x32xbf16, strided<[512, 1], offset: ?>>, %arg1: memref<16x32x2xbf16, strided<[64, 2, 1], offset: ?>>) {
+    %cst = arith.constant 0.000000e+00 : bf16
+    %c0 = arith.constant 0 : index
+    %expand_shape = memref.expand_shape %arg0 [[0, 1], [2]] output_shape [16, 2, 32] : memref<32x32xbf16, strided<[512, 1], offset: ?>> into memref<16x2x32xbf16, strided<[1024, 512, 1], offset: ?>>
+    %0 = vector.transfer_read %expand_shape[%c0, %c0, %c0], %cst {in_bounds = [true, true, true]} : memref<16x2x32xbf16, strided<[1024, 512, 1], offset: ?>>, vector<16x2x32xbf16>
+    %1 = vector.transpose %0, [0, 2, 1] : vector<16x2x32xbf16> to vector<16x32x2xbf16>
+    vector.transfer_write %1, %arg1[%c0, %c0, %c0] {in_bounds = [true, true, true]} : vector<16x32x2xbf16>, memref<16x32x2xbf16, strided<[64, 2, 1], offset: ?>>
+    return
+}
+
+// CHECK-LABEL: func.func @vnni_packing_0(
+// CHECK: %[[arg0:.*]]: memref<32x32xbf16, strided<[512, 1], offset: ?>>, %[[arg1:.*]]: memref<16x32x2xbf16, strided<[64, 2, 1], offset: ?>>) {
+// CHECK-DAG: %[[c28_i64:.*]] = arith.constant 28 : i64
+// CHECK-DAG: %[[c2_i64:.*]] = arith.constant 2 : i64
+// CHECK-DAG: %[[c32_i64:.*]] = arith.constant 32 : i64
+// CHECK-DAG: %[[c512_i64:.*]] = arith.constant 512 : i64
+// CHECK-DAG: %[[c0_i64:.*]] = arith.constant 0 : i64
+// CHECK: %[[dispatch:.*]] = call @xsmm_unary_dispatch(%[[c28_i64]], %[[c2_i64]], %[[c32_i64]], %[[c32_i64]], %[[c512_i64]], %[[c32_i64]], %[[c0_i64]])
+// CHECK: %[[expand_shape:.*]] = memref.expand_shape %[[arg0]] {{\[}}[0, 1], [2]] output_shape [16, 2, 32]
+// CHECK: %[[base_buffer:.*]], %[[offset:.*]], %[[sizes:.*]]:3, %[[strides:.*]]:3 = memref.extract_strided_metadata %expand_shape
+// CHECK-DAG:  %[[intptr:.*]] = memref.extract_aligned_pointer_as_index %[[expand_shape]]
+// CHECK-NEXT: %[[indexCast1:.*]] = arith.index_cast %[[intptr]]
+// CHECK-NEXT: %[[inttoptr:.*]] = llvm.inttoptr %[[indexCast1]]
+// CHECK: %[[base_buffer_0:.*]], %[[offset_1:.*]], %[[sizes_2:.*]]:3, %[[strides_3:.*]]:3 = memref.extract_strided_metadata %[[arg1]]
+// CHECK-DAG:  %[[intptr0:.*]] = memref.extract_aligned_pointer_as_index %[[arg1]]
+// CHECK-NEXT: %[[indexCast2:.*]] = arith.index_cast %[[intptr0]]
+// CHECK-NEXT: %[[inttoptr2:.*]] = llvm.inttoptr %[[indexCast2]]
+// CHECK:   call @xsmm_unary_invoke(%[[c2_i64]], %[[dispatch]], %[[inttoptr]], %[[offset]], %[[inttoptr2]], %[[offset_1]])
+
+// -----
+func.func @not_vnni_packing_1(%arg0: memref<32x32xf32, strided<[512, 1], offset: ?>>, %arg1: memref<16x32x2xf32, strided<[64, 2, 1], offset: ?>>) {
```

Comment:
took me a while to realize this was `f32`. Please update the function name with `not_vnni_in_f32` or something.

Reply:
Done.